src/components/input/input.js

```diff
@@ -650,7 +651,21 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       // Using the $validators for triggering the update works very well.
       renderCharCount();
 
-      return ( modelValue || element.val() || viewValue || '' ).length <= maxlength;
+      // Force the value into a string since it may be a number,
+      // which does not have a length property.
+      return String(modelValue || element.val() || viewValue || '').length <= maxlength;
+    };
+
+    ngModelCtrl.$validators['required'] = function(modelValue, viewValue) {
```

Comment:
This means that the default (or a custom) `required` validator will be overwritten 😱
So, for example `ctrl.$isEmpty()` (which is a standard way to define the behavior for the `required` validator), will be ignored.

This should at least be mentioned as a limitation in the docs. But it feels a little weird 😕
Is it really necessary? Why do you even need this here?

Reply:
This was required to get the tests to pass. It wasn't required for the actual validation and use case to work in manual testing. I think that I'll take a look at your comment above about just using the viewValue to see if that's already trimmed as I would like to avoid this if possible.